internal/server/server.go

```diff
@@ -94,13 +95,15 @@ func New(cfg Config) *Server {
 			FileLevel:     strings.ToLower(cfg.LogSettings.FileLevel),
 			FileLocation:  cfg.LogSettings.FileLocation,
 		}),
-		cfg:   cfg,
-		creds: credentials.NewStatic(cfg.S3Settings.AccessKeyID, cfg.S3Settings.SecretAccessKey, "", credentials.SignatureV4),
+		cfg:     cfg,
+		creds:   credentials.NewStatic(cfg.S3Settings.AccessKeyID, cfg.S3Settings.SecretAccessKey, "", credentials.SignatureV4),
+		metrics: newMetrics(),
 	}
 
 	s.getHostFn = s.getHost
 	s.srv.Handler = s.withRecovery(s.handler())
 	healthMux.HandleFunc("/health", s.healthHandler).Methods("GET")
```

Comment:
This is not just `healthMux` anymore. I'd change it to `serviceMux`, `serviceHost` and `serviceServer` as the variable names in various places.